src/goto-programs/json_goto_trace.cpp

```diff
@@ -114,6 +106,13 @@ void convert(
             type_string=from_type(ns, identifier, symbol->type);
 
           json_assignment["mode"]=json_stringt(id2string(symbol->mode));
+          assert(step.full_lhs_value.is_not_nil());
+          full_lhs_value=json(step.full_lhs_value, ns, symbol->mode);
+        }
+        else
+        {
+          if(step.full_lhs_value.is_not_nil())
```

Comment:
Actually I found one assertion failing because of the assertion `lang && "unknown mode"` I added in `array-refinement/Array_UF16` but it's not a test that is normally activated for `CI`